community/lucene-index/src/test/java/org/neo4j/index/LegacyIndexTest.java

```diff
@@ -139,6 +142,114 @@ public void shouldThrowIllegalArgumentChangingTypeOfFieldOnNodeIndex()
     }
 
     @Test
+    public void shouldBeAbleToAddNodesAfterRemovalOfKey()
+    {
+        String indexName = "index";
+        long nodeId;
+        //add two keys and delete one of them
+        try ( Transaction tx = db.beginTx() )
+        {
+            Node node = db.createNode();
+            nodeId = node.getId();
+            Index<Node> nodeIndex = db.index().forNodes( indexName );
+            nodeIndex.add( node, "key", "hej" );
+            nodeIndex.add( node, "keydelete", "hej" );
+            tx.success();
+        }
+
+        try ( Transaction tx = db.beginTx() )
+        {
+            Index<Node> nodeIndex = db.index().forNodes( indexName );
+            nodeIndex.remove( db.getNodeById( nodeId ), "keydelete" );
+            tx.success();
+        }
+
+        db.shutdownAndKeepStore();
+        db.getGraphDatabaseAPI();
+
+        //should be able to add more stuff to the index
+        try ( Transaction tx = db.beginTx() )
+        {
+            Node node = db.createNode();
+            Index<Node> nodeIndex = db.index().forNodes( indexName );
+            nodeIndex.add( node, "key", "hej" );
```

Comment:
did you mean to use `keydelete` here as key?

Reply:
No, I want to verify the that the indextype of `key` is correct.